stack/services/src/main/java/org/apache/usergrid/services/assets/data/S3BinaryStore.java

```diff
@@ -134,10 +134,11 @@ public void write( final UUID appId, final Entity entity, InputStream inputStrea
         }
         else { // bigger than 5mb... dump 5 mb tmp files and upload from them
 
-            // todo: yes, AsyncBlobStore is deprecated, but there appears to be no replacement yet
-            final AsyncBlobStore blobStore = getContext().getAsyncBlobStore();
+            // create temp file and copy entire file to that temp file
 
-            File tempFile = File.createTempFile( entity.getUuid().toString(), "tmp" );
+            LOG.debug( "Writing temp file for S3 upload" );
```

Comment:
Todd, it appears that we (unintentionally?) limit file size to 10 mb, because we read 5pm and then on this line we only only read up to 5 pm more: https://goo.gl/sf0Tof